lib/servers/Pulser2/sequence.py

```diff
@@ -71,7 +71,7 @@ def _addNewSwitch(self, timeStep, chan, value):
         if self.switchingTimes.has_key(timeStep):
             if self.switchingTimes[timeStep][chan]: # checks if 0 or 1/-1
                 # if set to turn off, but want on, replace with zero, fixes error adding 2 TTLs back to back
-                if self.switchingTimes[timeStep][chan] == -1 and value == 1:
+                if self.switchingTimes[timeStep][chan] * value == -1:
```

Comment:
This logic is getting difficult to parse efficiently.  We should consider creating a SwitchTime class instead of using a switchingTimes dictionary.  Will add more capabilities that will make reading the code much easier.

Reply:
I'm not sure I agree. It's only two or three more lines than the original code. Maybe we could leave that as a later project?

Reply:
I think replacing the dictionary would require extensive effort and we need this fix now so I agree with @theoriginaljuice we should move forward and think about this later

Reply:
I would merge now.  I was commenting on the overall readability (triply nested logic statements and a complex dictionary), not the change.